### DISCUSSION TITLE
Create crawlers automatically using the existing templates

### DISCUSSION THREAD

**Discussion by dipu-bd** [lncrawl/lightnovel-crawler#1673] [Announcements]
After the new changes to `dev` now it is possible to create crawlers automatically using the existing templates.

A guide to how to do it:

```
(venv) $ python -m lncrawl --bot lookup
================================================================================
╭╮╱╱╱╱╱╱╭╮╱╭╮╱╱╱╱╱╱╱╱╱╱╱╱╭╮╱╭━━━╮╱╱╱╱╱╱╱╱╱╭╮
┃┃╱╱╱╱╱╱┃┃╭╯╰╮╱╱╱╱╱╱╱╱╱╱╱┃┃╱┃╭━╮┃╱╱╱╱╱╱╱╱╱┃┃
┃┃╱╱╭┳━━┫╰┻╮╭╋━╮╭━━┳╮╭┳━━┫┃╱┃┃╱╰╋━┳━━┳╮╭╮╭┫┃╭━━┳━╮
┃┃╱╭╋┫╭╮┃╭╮┃┃┃╭╮┫╭╮┃╰╯┃┃━┫┃╱┃┃╱╭┫╭┫╭╮┃╰╯╰╯┃┃┃┃━┫╭╯
┃╰━╯┃┃╰╯┃┃┃┃╰┫┃┃┃╰╯┣╮╭┫┃━┫╰╮┃╰━╯┃┃┃╭╮┣╮╭╮╭┫╰┫┃━┫┃
╰━━━┻┻━╮┣╯╰┻━┻╯╰┻━━╯╰╯╰━━┻━╯╰━━━┻╯╰╯╰╯╰╯╰╯╰━┻━━┻╯
╱╱╱╱╱╭━╯┃ v3.0.0
╱╱╱╱╱╰━━╯ 🔗 https://github.com/dipu-bd/lightnovel-crawler
--------------------------------------------------------------------------------

➡ Press  Ctrl + C  to exit

? Enter novel page url: https://www.novelmtl.com/novel/ancient-true-dragon-art.html
🍀 Checking MadaraTemplate [1 of 2] 
  ➡ Create instance : success
  ➡ initialize() : success
  ➡ read_novel_info() : failed
    No title found

🍀 Checking NovelMTLTemplate [2 of 2] 
  ➡ Create instance : success
  ➡ initialize() : success
  ➡ read_novel_info() : success
  ➡ download_chapter_body() : success

? Enter language: en
? Does it contain Manga/Manhua/Manhwa? No
? Does it contain Machine Translations? No

📦 Generated source file 📦
➡ /home/dipu/projects/lightnovel-crawler/sources/en/n/novelmtl.py
```

If you are confused about the `language`, check the [language_codes](https://github.com/dipu-bd/lightnovel-crawler/blob/77fcd3fccadddbdc3c3d411407a336b4234aa1af/lncrawl/assets/languages.py).

**Comment by mygithub-acct** [comment]
If this didn't work and I manually try to create my own source, where should I copy the file to? The normal source folder no longer exist in the lncrawl folder like it used to be.
![image](https://user-images.githubusercontent.com/71422313/195011964-230ec16f-d9cc-4092-89d7-97c64a3c0222.png)

**Reply by dipu-bd** [reply]
check one level above it. i recommend cloning the repository and edit there. send a Pull Request once you are done

**Reply by mygithub-acct** [reply]
Found it, thanks! Copied my file into the folder however lncrawl doesn't recognise the new website because it doesn't know there is a new file in sources folder. How can I trigger it to update the list of sources? Sorry, lots of newbie questions
